qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -241,24 +240,13 @@ def __init__(
                   tuple (label, coeff), or list [(label, coeff)].
             register_length: positive integer that represents the length of registers.
             display_format: If sparse, the label is represented sparsely during output.
-                            if dense, the label is represented densely during output. (default: dense)
+                            if dense, the label is represented densely during output. (default: sparse)
 
         Raises:
             ValueError: given data is invalid value.
             TypeError: given data has invalid type.
         """
-        if display_format is None:
-            display_format = "dense"
-            if FermionicOp._display_format_warn:
-                FermionicOp._display_format_warn = False
-                warnings.warn(
-                    "The default value for `display_format` will be changed from 'dense' "
-                    "to 'sparse' in version 0.3.0. Once that happens, you must specify "
-                    "display_format='dense' directly.",
-                    stacklevel=2,
-                )
-
-        self.display_format = display_format
+        self.display_format = display_format or "sparse"
```

Comment:
Is there a reason not to remove Optional now from display format and just have it of type `str` and default it right there to `"sparse"`. It was Optional I believe so we could detect if a value was supplied by the user or not and if not (ie was None) raise a message about the default changing.